airbyte-ci/connectors/pipelines/Python connector images analysis.md

```diff
@@ -0,0 +1,738 @@
+
+
+
+#### Differences in `TRACEPARENT` and `OTEL_TRACE_PARENT`
+On these env var we have a different value for each connector image.
+I believe these are set at build time by the Docker engine.
+These Ids can be used for tracing. I don't know if they are actively use in our current infrastructure.
+
+**Takeaways**:
+On connector build in `airbyte-ci` we should verify that these env var are set.
+
+
+```python
+get_unique_values_for_env_vars(docker_variants_env_vars)["TRACEPARENT"]
+```
+
+
+
+
+    {'00-05babbaa467d2b7101a1c3c4b75c1b5f-dd8adab127d27f39-01',
+     '00-15262d0156d304755c1a37aab7400d91-d000781658986ada-01',
+     '00-492c7d03978cbd27941a0f01c64d0ea2-53174fa4a8f396d5-01',
+     '00-4cfe6e40354d22eef1c8bd44a53de77b-c791eac4904fd1f2-01',
+     '00-5a8d6c04188be1aebce946a1ef5e61a2-52f2d30a2ae782ed-01',
+     '00-624b92c81c4de6a0f0f8e7bdee0e3ba3-0ab1ead787e1426b-01',
+     '00-659529510ea31fffdb43e578b7a61e85-3274b696e496d0d2-01',
+     '00-7cd60817e3b3542b6bd91c22098ff33a-03a4d5e08021248d-01',
+     '00-9d69486c8f84d271449e4885eec67943-7c2a3f817dbca68b-01',
+     '00-a215dd2bf43039a9f04c28219a7cb5e9-5c12bf9a7651cfba-01',
+     '00-af060a2eb1bd83b31d40d0da6232e6e1-d1e86c8724866a37-01',
+     '00-caf0d2479373ae86133704e4e2f19e73-3536098ec356a6c0-01',
+     '00-fca418ed06415f3741ff550a232cf257-0216c5a3245cbf48-01',
+     '00-fff1fd9d58a2ba8c407a0ee94c965aa5-eafd1f6498b40fac-01'}
+
+
+
+
+```python
+get_unique_values_for_env_vars(docker_variants_env_vars)["OTEL_TRACE_PARENT"]
+```
+
+
+
+
+    {'00-05babbaa467d2b7101a1c3c4b75c1b5f-dd8adab127d27f39-01',
+     '00-15262d0156d304755c1a37aab7400d91-d000781658986ada-01',
+     '00-492c7d03978cbd27941a0f01c64d0ea2-53174fa4a8f396d5-01',
+     '00-4cfe6e40354d22eef1c8bd44a53de77b-c791eac4904fd1f2-01',
+     '00-5a8d6c04188be1aebce946a1ef5e61a2-52f2d30a2ae782ed-01',
+     '00-624b92c81c4de6a0f0f8e7bdee0e3ba3-0ab1ead787e1426b-01',
+     '00-659529510ea31fffdb43e578b7a61e85-3274b696e496d0d2-01',
+     '00-7cd60817e3b3542b6bd91c22098ff33a-03a4d5e08021248d-01',
+     '00-9d69486c8f84d271449e4885eec67943-7c2a3f817dbca68b-01',
+     '00-a215dd2bf43039a9f04c28219a7cb5e9-5c12bf9a7651cfba-01',
+     '00-af060a2eb1bd83b31d40d0da6232e6e1-d1e86c8724866a37-01',
+     '00-caf0d2479373ae86133704e4e2f19e73-3536098ec356a6c0-01',
+     '00-fca418ed06415f3741ff550a232cf257-0216c5a3245cbf48-01',
+     '00-fff1fd9d58a2ba8c407a0ee94c965aa5-eafd1f6498b40fac-01'}
+
+
+
+### Connector specific environment variables
+Some connector are setting custom environment variables.
+#### `AIRBYTE_IMPL_MODULE` and `AIRBYTE_IMPL_PATH` 
+These variables are [used in the CDK](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/entrypoint.py#L256) to override some default behavior in the connector entrypoint.  `source-zendesk-chat` is setting these env var. I can't assess if it's rightfully doing so. We'll support this by custom pre/post build hooks for this connector.
```

Comment:
I don't think it should be a blocker that we'd have to wait for this PR on, but based on how long this project might take, I'd suggest that we open a ticket for resolving issues in `source-zendesk-chat` to see if we can get it to conform, since we'd have to put in work to add the custom hooks anyway. This is something that could be easily parallelized - our team could take it on too.